frontend/src/app/services/auth.service.ts

```diff
@@ -53,12 +49,17 @@ export class AuthService {
 
   private handleLoginSuccess(response: any) {
     const token = response.access_token;
+
+    if (!token) {
+      this.loginError = 'Login failed due to a missing access token';
+    }
+
     localStorage.setItem('token', token);
     this.router.navigate(['/manage']);
   }
 
-  private handleLoginError(error: any): void {
+  private async handleLoginError(error: any): Promise<void> {
```

Comment:
Apparently this needs to be `async` when I switch to `catchError()` (And `Promise< >` added because async functions need to return a promise of whatever it is you're returning)